gqlalchemy/instance_runner.py

```diff
@@ -0,0 +1,228 @@
+        process.
+    """
+
+    def __init__(self, binary_path: str = "/usr/lib/memgraph/memgraph", user: str = "", **data) -> None:
+        super().__init__(**data)
+        self.binary_path = binary_path
+        self.user = user
+
+    def start(self, restart: bool = False) -> "Memgraph":
+        """Start the Memgraph instance and return the connection object.
+
+        Attributes:
+            restart: A bool indicating if the instance should be
+              restarted if it's already running.
+        """
+        if not restart and self.is_running():
+            return
+        self.stop()
+
+        args_mg = f"{self.binary_path } " + (" ").join([f"{k}={v}" for k, v in self.config.items()])
+        if self.user != "":
+            args_mg = f"sudo runuser -l {self.user} -c '{args_mg}'"
+        self.proc_mg = subprocess.Popen(args_mg, shell=True)
+        wait_for_port(self.host, self.port)
+
+        self.connect()
+        return self.memgraph
+
+    def stop(self) -> int:
+        """Stop the Memgraph instance."""
+        if not self.is_running():
+            return
+        procs = set()
+        process = psutil.Process(self.proc_mg.pid)
+        procs.add(process)
+        for proc in process.children(recursive=True):
+            procs.add(proc)
+            os.system(f"sudo kill {proc.pid}")
+        process.kill()
+        psutil.wait_procs(procs)
+
+    def is_running(self) -> bool:
+        """Check if the Memgraph instance is still running."""
+        if self.proc_mg is None:
+            return False
+        if self.proc_mg.poll() is not None:
+            return False
+        return True
+
+
+class MemgraphInstanceDocker(MemgraphInstance):
+    """A class for managing Memgraph instances started in Docker containers.
+
+    Attributes:
+      docker_image: An enum representing the Docker image. Values:
+        `DockerImage.MEMGRAPH` and `DockerImage.MAGE`.
+      docker_image_tag: A string representing the tag of the Docker image.
+    """
+
+    def __init__(
+        self, docker_image: DockerImage = DockerImage.MEMGRAPH, docker_image_tag: str = "latest", **data
+    ) -> None:
+        super().__init__(**data)
+        self.docker_image = docker_image
+        self.docker_image_tag = docker_image_tag
+        self._client = docker.from_env()
+        self._container = None
+
+    def start(self, restart: bool = False) -> "Memgraph":
+        """Start the Memgraph instance and return the connection object.
+
+        Attributes:
+            restart: A bool indicating if the instance should be
+              restarted if it's already running.
+        """
+        if not restart and self.is_running():
+            return
+        self.stop()
+        self._container = self._client.containers.run(
+            image=self.docker_image.value + ":" + self.docker_image_tag,
+            command="/usr/lib/memgraph/memgraph " + (" ").join([f"{k}={v}" for k, v in self.config.items()]),
```

Comment:
I would suggest promoting all strings to string constants

Reply:
The paths and error messages have been replaced with constants.